javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js

```diff
@@ -1,272 +1,259 @@
+  const sumOfAllDices = displayArray.reduce((a, b) => a + b, 0);
 
-        return true;
-      }
-    }
-    alert("Please select a score.");
-    return false;
+  // Return the highest duplicates only if there are at least three or four of them
+  if (highestCount >= 4) {
+    updateRadioOption(1, sumOfAllDices);
+  } else if (highestCount >= 3) {
+    updateRadioOption(0, sumOfAllDices);
+  } else {
+    updateRadioOption(5, 0);
   }
+};
 
-  gameOver() {
-    if (
-      confirm(
-        `You scored ${game.totalScore} points. Do you want to play again?`
-      )
-    )
-      location.reload();
-  }
-}
+/**
+ * ! This will help us detect full house
+ */
+
+const detectFullHouse = (arr) => {
+  // condiiton, 3 of a kind and 2 of a kind
+  const counts = {};
 
-const game = new Game();
+  // Count the occurrences of each number in the array
+  for (const num of arr) {
+    counts[num] = counts[num] ? counts[num] + 1 : 1;
+  }
 
-rollDiceBtn.addEventListener("click", (e) => {
-  if (game.currentRound <= 6) {
-    game.rollDice();
-    game.resetRadioInputs();
-    game.generateValidScoreOptions();
-    game.enableValidScoreInputs();
-    game.updateStatsUI();
-    keepScoreBtn.disabled = false;
+  // Check if there are three of one number and two of another number
+  let hasThreeOfAKind = false;
+  let hasPair = false;
 
-    if (game.rollsInCurrentRound % 3 === 0) {
-      e.target.disabled = true;
+  for (const num in counts) {
+    if (counts[num] === 3) {
+      hasThreeOfAKind = true;
+    } else if (counts[num] === 2) {
+      hasPair = true;
     }
   }
-});
 
-keepScoreBtn.addEventListener("click", (e) => {
-  if (game.currentRound <= 6 && game.isKeepScoreSuccess()) {
-    game.resetRadioInputs();
-    game.updateStatsUI();
-    game.updateScoreUI();
-    rollDiceBtn.disabled = false;
-    e.target.disabled = true;
+  if (hasThreeOfAKind && hasPair) {
+    updateRadioOption(2, 25);
+  } else {
+    updateRadioOption(5, 0);
   }
+};
+
+/**
+ * ! This will help us run all the functions that detect the outcome of the roll
+ */
+
+const findRollResult = (arr) => {
+  straightDetector(arr);
+  getHighestDuplicates(arr);
+  detectFullHouse(arr);
+};
+
+/**
+ * ! This function will help us reset the game
+ */
+
+const resetGame = () => {
+  displayArray = [0, 0, 0, 0, 0];
+  score = 0;
+  totalScore = 0;
+  round = 1;
+  rolls = 0;
 
-  if (game.currentRound > 6) {
-    // Give DOM some time to complete manipulation then check & run gameOver
-    setTimeout(game.gameOver, 100);
+  diceButtons.forEach((dice, index) => {
+    dice.textContent = displayArray[index];
+  });
+
+  totalScoreText.innerHTML = totalScore;
+  scoreHistory.innerHTML = "";
+
+  currentRoundRollsText.innerHTML = rolls;
+  currentRoundText.innerHTML = round;
+
+  resetRadioOption();
+};
+
+/**
+ * ! this is the trigger for the app to work
+ */
+
+rollDiceBtn.addEventListener("click", () => {
+  if (rolls === 3) {
+    window.alert("Please select an option");
```

Comment:
```suggestion
    window.alert("You have made three rolls this round. Please select a score.");
```

Clarify the message.